src/Cake.Issues.Markdownlint/LogFileFormat/MarkdownlintLogfileHelper.cs

```diff
@@ -0,0 +1,39 @@
+﻿namespace Cake.Issues.Markdownlint.LogFileFormat
+{
+    using System;
+    using System.Text.RegularExpressions;
+
+    /// <summary>
+    /// Helper class for Markdownlint log files.
+    /// </summary>
+    internal static class MarkdownlintLogFileHelper
+    {
+        /// <summary>
+        /// Reads the RegEx for a Mardown issue line depending its line format.
+        /// </summary>
+        /// <param name="markdownIssueLine">Markdown issue line.</param>
+        /// <returns>The RegEx pattern to parse the Markdown issue line.</returns>
+        public static Regex GetRegexForMarkdownIssueLine(string markdownIssueLine)
+        {
+            markdownIssueLine.NotNull(nameof(markdownIssueLine));
+
+            // RegEx pattern to parse an Markdown issue line starting with Markdownlint-cli version >= 0.19.0
+            var regex = new Regex(@"(.*):(\d*) (MD\d*)/((?:\w*-*/*)*) (.*)");
+            var regexMatch = regex.Match(markdownIssueLine);
+
+            if (!regexMatch.Success)
+            {
+                // RegEx pattern to parse an Markdown issue line Markdownlint-cli version < 0.19.0
+                regex = new Regex(@"(.*): (\d*): (MD\d*)/((?:\w*-*/*)*) (.*)");
+                regexMatch = regex.Match(markdownIssueLine);
+            }
+
+            if (regexMatch.Success)
+            {
+                return regex;
+            }
+
+            throw new ArgumentException("No RegEx found for the Markdown issue line.");
```

Comment:
This should not be an `ArgumentException`. Use `InvalidOperationException` (or a custom exception)